rewrite-java/src/test/java/org/openrewrite/java/RemoveMethodInvocationsVisitorTest.java

```diff
@@ -492,4 +493,106 @@ public void method() {
           )
         );
     }
+
+    @Test
+    void removeStaticMethodFromImport() {
+        rewriteRun(
+          spec -> spec.recipe(createRemoveMethodsRecipe("org.junit.jupiter.api.Assertions assertTrue(..)")),
+          // language=java
+          java(
+            """
+              import static org.junit.jupiter.api.Assertions.assertTrue;
+              
+              class Test {
+                  void method() {
+                      assertTrue(true);
+                  }
+              }
+              """,
+            """
+              class Test {
+                  void method() {
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void removeStaticMethodFromImportWithAssignment() {
+        rewriteRun(
+          spec -> spec.recipe(createRemoveMethodsRecipe("java.util.Collections emptyList()")),
+          // language=java
+          java(
+            """
+              import static java.util.Collections.emptyList;
+              
+              class Test {
+                  void method() {
+                      List<Object> emptyList = emptyList();
+                  }
+              }
+              """,
+            """
+              class Test {
+                  void method() {
+                      List<Object> emptyList;
```

Comment:
As this is just a utility visitor and not a recipe, I think it would not necessarily be a big issue if it produces code that isn't quite correct, as long as what it is doing is expected by the calling recipe. The recipe would be responsible for making additional changes to make the code correct. But this is exactly my worry with this visitor. I am not sure it is clear enough what exactly it does and what that entails.

Removing all static method calls which are statements (where the return type is `void` or the method is called in a statement and the return value is ignored) seems safe enough and from what I gather what the intention is here for the EasyMock recipes. But is that what the purpose is of this visitor? Should there maybe be a separate visitor / recipe for that?

Reply:
> Removing all static method calls which are statements (where the return type is void or the method is called in a statement and the return value is ignored) seems safe enough.
I agree, that's exactly what you would expect if you ask me. Which is actually the same as I proposed, to not do anything when the method invocation is assigned to a variable.

>But is that what the purpose is of this visitor? Should there maybe be a separate visitor / recipe for that?

Reply:
We could remove any method invocations that are used as statements, which would cover what is requested here.

Reply:
That's what I did: https://github.com/openrewrite/rewrite/blob/b6e42da3c42402f34c03c90595c8551734bad4ad/rewrite-java/src/main/java/org/openrewrite/java/RemoveMethodInvocationsVisitor.java#L80 😉.